programs/mpl-core/src/plugins/lifecycle.rs

```diff
@@ -654,6 +807,124 @@ pub(crate) trait PluginValidation {
     ) -> Result<ValidationResult, ProgramError> {
         abstain!()
     }
+
+    /// Process side effects for the add plugin lifecycle action.
+    /// This gets called on all existing plugins when a new plugin is added.
+    fn side_effects_add_plugin(
+        &self,
+        _ctx: &PluginSideEffectContext,
+    ) -> Result<Plugin, ProgramError> {
+        unreachable!()
+    }
```

Comment:
These aren't really unreachable are they?  Its more like you are just trying to do a harsh assert if the default trait method is called?  If that's the case I think you can just list the trait method without a default